src/services/findAllReferences.ts

```diff
@@ -622,7 +622,15 @@ namespace ts.FindAllReferences {
             // Could not find a symbol e.g. unknown identifier
             if (!symbol) {
                 // String literal might be a property (and thus have a symbol), so do this here rather than in getReferencedSymbolsSpecial.
-                return !options.implementations && isStringLiteralLike(node) ? getReferencesForStringLiteral(node, sourceFiles, checker, cancellationToken) : undefined;
+                if (!options.implementations && isStringLiteralLike(node)) {
+                    const refFileMap = program.getRefFileMap();
+                    const referencedFileName = node.getSourceFile().resolvedModules?.get(node.text)?.resolvedFileName;

```

Comment:
do you not need to check if this is really module literal?( i am not sure if  <reference type="type"/> enters this path or reference path=`reference` either but if it does you would need to distinguish that right ?

Reply:
I wanted to avoid exhaustively listing supported syntaxes, but you’re right, this triggered a false positive on random string literals that exactly matched legitimate file references elsewhere in the same file. Fixed now.